### PR TITLE
Simplify exception dispatch to branches

### PR DESCRIPTION
**Pull request by aheejin**
This proposes to use the idea suggested by #58 on using `br_on_exn` instead of `if_except`.

### CODE REVIEW

File: proposals/Exceptions.md [hunk 1]
```diff
@@ -215,39 +216,57 @@ exception on top of the stack is popped and then thrown.
...
+does not match, the `except_ref` value remains on the stack. For example, when
+an `except_ref` contains an exception of type (i32 i64), the target block
+signature should be (i32 i64) as well, as in the following example:
 
-if_except block_type except_index
-  Instruction*
-else
-  Instruction*
+```
+block $l (result i32 i64)
```
Comment (aheejin) [inline proposals/Exceptions.md:240]: Even though #58 suggested the multi-value proposal is a prerequisite for this and this example also uses a multi-value return block, the multi-value proposal is not necessarily a prerequisite for this exception handling proposal; we can do with a single value yielding blocks until the multi value proposal arrives.

File: proposals/Exceptions.md [hunk 2]
```diff
@@ -128,9 +128,10 @@ Exception indices are used by:
 1. The `throw` instruction which creates a WebAssembly exception with the
    corresponding exception tag, and then throws it.
 
-2. The `if_except` instruction queries an exception to see if the corresponding
-   exception tag denoted by the exception index. If true it pushes the
-   corresponding values of the exception onto the stack.
+2. The `br_on_exn` instruction queries an exception to see if the corresponding
```
Comment (rossberg) [inline proposals/Exceptions.md]: "to see if the" -> "to see if it matches the"

File: proposals/Exceptions.md [hunk 3]
```diff
@@ -128,9 +128,10 @@ Exception indices are used by:
...
+   exception tag denoted by the exception index. If true it branches to the
+   given label and pushes the corresponding values of the exception onto the
```
Comment (rossberg) [inline proposals/Exceptions.md]: "value" -> "argument values" for clarity?

File: proposals/Exceptions.md [hunk 4]
```diff
@@ -215,39 +216,57 @@ exception on top of the stack is popped and then thrown.
...
-  Instruction*
-end
+br_on_exn label except_index
+```
+
+The `br_on_exn` instruction checks the exception tag of an `except_ref` on top
+of the stack if it matches the given exception index. If it does, it branches
+out to the label referenced by the instruction (In the binary form, the label
+will be converted to a relative depth immediate, like other branch
+instructions), and while doing that, pops exception values from the `except_ref`
```
Comment (rossberg) [inline proposals/Exceptions.md]: "pops exception values" -> "pops the `except_ref` value from the stack and instead pushes the exception's argument values"

**Review by rossberg — APPROVED** [review]
Looks good modulo some wording tweaks.

File: proposals/Exceptions.md (hunk 2 above)
Comment (aheejin) [reply proposals/Exceptions.md]: Done

File: proposals/Exceptions.md (hunk 3 above)
Comment (aheejin) [reply proposals/Exceptions.md]: Done

File: proposals/Exceptions.md (hunk 4 above)
Comment (aheejin) [reply proposals/Exceptions.md]: Done

**Comment by mstarzinger** [comment]
Looks good. Thanks!

**Pull request merged**